components/terrain/cellborder.cpp

```diff
@@ -76,23 +76,24 @@ void CellBorder::createCellBorderGeometry(int x, int y)
     mCellBorderNodes[std::make_pair(x,y)] = borderGeode;
 }
 
-void CellBorder::destroyCellBorderGeometry(int x, int y)
+CellBorder::CellGrid::iterator CellBorder::destroyCellBorderGeometry(int x, int y)
 {
     CellGrid::iterator it = mCellBorderNodes.find(std::make_pair(x,y));
 
     if (it == mCellBorderNodes.end())
-        return;
+        return mCellBorderNodes.end();
 
     osg::ref_ptr<osg::Node> borderNode = it->second;
     mRoot->removeChild(borderNode);
 
-    mCellBorderNodes.erase(it);
+    return mCellBorderNodes.erase(it);
 }
 
 void CellBorder::destroyCellBorderGeometry()
 {
-    for (CellGrid::iterator it = mCellBorderNodes.begin(); it != mCellBorderNodes.end(); ++it)
-        destroyCellBorderGeometry(it->first.first,it->first.second);
+    CellGrid::iterator it = mCellBorderNodes.begin();
+    while (it != mCellBorderNodes.end())
+        it = destroyCellBorderGeometry(it->first.first,it->first.second);
```

Comment:
This implementation will call `map::find` to get the iterator that is already here. Just increases complexity from `O(n)` to `O(log n)`. I suggest to separate these operations. Leave single node `destroyCellBorderGeometry` as is but change for all to this:
```c++
void CellBorder::destroyCellBorderGeometry()
{
    for (const auto& v : mCellBorderNodes)
        mRoot->removeChild(v.second);
    mCellBorderNodes.clear();
}
```
It's much clear that we want to remove all of them. Call erase on each remove also may have an additional cost.

Reply:
Done.